docs/Chipyard-Basics/Configs-Parameters-Mixins.rst

```diff
@@ -74,52 +74,37 @@ Cake Pattern
 A cake pattern is a Scala programming pattern, which enable "mixing" of multiple traits or interface definitions (sometimes referred to as dependency injection).
 It is used in the Rocket Chip SoC library and Chipyard framework in merging multiple system components and IO interfaces into a large system component.
 
-This example shows a Rocket Chip based SoC that merges multiple system components (BootROM, UART, etc) into a single top-level design.
+This example shows the Chipyard default top that composes multiple traits together into a fully-featured SoC with many optional components.
+
+
+.. literalinclude:: ../../generators/chipyard/src/main/scala/Top.scala
+    :language: scala
+    :start-after: DOC include start: Top
+    :end-before: DOC include end: Top
 
-.. _cake-example:
-.. code-block:: scala
 
-  class MySoC(implicit p: Parameters) extends RocketSubsystem
-    with CanHaveMasterAXI4MemPort
-    with HasPeripheryBootROM
-    with HasNoDebug
-    with HasPeripherySerial
-    with HasPeripheryUART
-    with HasPeripheryIceNIC
-  {
-     lazy val module = new MySoCModuleImp(this)
-  }
-
-  class MySoCModuleImp(outer: MySoC) extends RocketSubsystemModuleImp(outer)
-    with CanHaveMasterAXI4MemPortModuleImp
-    with HasPeripheryBootROMModuleImp
-    with HasNoDebugModuleImp
-    with HasPeripherySerialModuleImp
-    with HasPeripheryUARTModuleImp
-    with HasPeripheryIceNICModuleImp
-
-There are two "cakes" here. One for the lazy module (ex. ``HasPeripherySerial``) and one for the lazy module
-implementation (ex. ``HasPeripherySerialModuleImp`` where ``Imp`` refers to implementation). The lazy module defines
+There are two "cakes" here. One for the lazy module (ex. ``CanHavePeripherySerial``) and one for the lazy module
+implementation (ex. ``CanHavePeripherySerialModuleImp`` where ``Imp`` refers to implementation). The lazy module defines
 all the logical connections between generators and exchanges configuration information among them, while the
 lazy module implementation performs the actual Chisel RTL elaboration.
 
-In the ``MySoC`` example class, the "outer" ``MySoC`` instantiates the "inner"
-``MySoCModuleImp`` as a lazy module implementation. This delays immediate elaboration
+In the ``Top`` example class, the "outer" ``Top`` instantiates the "inner"
+``TopModule`` as a lazy module implementation. This delays immediate elaboration
 of the module until all logical connections are determined and all configuration information is exchanged.
-The ``RocketSubsystem`` outer base class, as well as the
-``HasPeripheryX`` outer traits contain code to perform high-level logical
-connections. For example, the ``HasPeripherySerial`` outer trait contains code
-to lazily instantiate the ``SerialAdapter``, and connect the ``SerialAdapter``'s
+The ``Syatem`` outer base class, as well as the
+``CanHavePeripheryX`` outer traits contain code to perform high-level logical
```

Comment:
Nit: To make this clearer to read, maybe have the X separated out somehow i.e. `CanHavePeriphery<X>`

Reply:
Done